scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -183,26 +183,6 @@ class RunCoverage(
   }
 
   private fun runCoverageForFile(filePath: String): CoverageReport {
-    val exemption = testFileExemptionList[filePath]
-    if (exemption != null && exemption.testFileNotRequired) {
-      return CoverageReport.newBuilder()
-        .setExemption(
-          CoverageExemption.newBuilder()
-            .setFilePath(filePath)
-            .build()
-        ).build()
-    } else {
-      val testFilePaths = findTestFiles(rootDirectory, repoRoot, filePath)
-      if (testFilePaths.isEmpty()) {
-        return CoverageReport.newBuilder()
-          .setFailure(
-            CoverageFailure.newBuilder()
-              .setFilePath(filePath)
-              .setFailureMessage("No appropriate test file found for $filePath.")
-              .build()
-          ).build()
-      }
```

Comment:
Why is this being removed now?

Reply:
They do exist but with additional checks for IncompatibleSources, I thought it would be better to have them in when cases so refactored them, 

[**RunCoverage.kt**](https://github.com/oppia/oppia-android/pull/5465/files#diff-62c81482e7c4b720da2d4efc73a48d10369eee097e0bf3439def6648ff150c23R188)
```
return when {
      exemption?.testFileNotRequired == true -> {
        CoverageReport.newBuilder()
          .setExemption(
            CoverageExemption.newBuilder()
              .setFilePath(filePath)
              .setExemptionReason(
                "This file is exempted from having a test file; " +
                  "skipping coverage check."
              )
              .build()
          ).build()
      }
      exemption?.sourceFileIsIncompatibleWithCodeCoverage == true -> {
        CoverageReport.newBuilder()
          .setExemption(
            CoverageExemption.newBuilder()
              .setFilePath(filePath)
              .setExemptionReason(
                "This file is incompatible with code coverage tooling; " +
                  "skipping coverage check."
              )
              .build()
          ).build()
      }
      else -> {
        val testFilePaths = findTestFiles(rootDirectory, repoRoot, filePath)
        when {}
     }
....
}
```

Reply:
Ah thanks for clarifying!